src/torchmetrics/detection/helpers.py

```diff
@@ -11,20 +11,26 @@
 # WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 # See the License for the specific language governing permissions and
 # limitations under the License.
-from typing import Dict, Sequence
+from typing import Dict, List, Literal, Sequence, Union
 
 from torch import Tensor
 
 
 def _input_validator(
-    preds: Sequence[Dict[str, Tensor]], targets: Sequence[Dict[str, Tensor]], iou_type: str = "bbox"
+    preds: Sequence[Dict[str, Tensor]],
+    targets: Sequence[Dict[str, Tensor]],
+    iou_type: Union[Literal["bbox", "segm"], List[str]] = "bbox",
 ) -> None:
     """Ensure the correct input format of `preds` and `targets`."""
-    if iou_type == "bbox":
-        item_val_name = "boxes"
-    elif iou_type == "segm":
-        item_val_name = "masks"
-    else:
+    if not isinstance(iou_type, list):
+        iou_type = [iou_type]
+
+    item_val_name = []
+    if "bbox" in iou_type:
+        item_val_name.append("boxes")
+    if "segm" in iou_type:
+        item_val_name.append("masks")
+    if any(i not in ["bbox", "segm"] for i in iou_type):
         raise Exception(f"IOU type {iou_type} is not supported")
```

Comment:
```suggestion
    if any(i not in ["bbox", "segm"] for i in iou_type):
        raise Exception(f"IOU type {iou_type} is not supported")
    item_val_name = []
    if "bbox" in iou_type:
        item_val_name.append("boxes")
    if "segm" in iou_type:
        item_val_name.append("masks")
```

Doesnt matter much but it would be best to first check if there are incorrect iou type and then do the rest.